c7n/tags.py

```diff
@@ -1085,7 +1082,7 @@ def process_resource(self, client, r, related_tags, tag_keys, tag_action):
         if not tags:
             return
         if not isinstance(tag_action, UniversalTag):
```

Comment:
I believe this is where I ran into problems that started this whole thing.  Basically, if the resource's `tag_action` is a UniversalTag, it passed the tags in list form, otherwise it passes it in dict form.  Maybe there was a resource that returned tags in dict form and we didn't have an augment step to convert it to our standard list form?  or it was related to other changes I was trying to write where we would more generically be calling `process_resource_set()` and didn't want to have to care which form it was in.  It's also difficult to keep straight which form a particular function expects.  In most cases, it takes the tag argument and passes it straight to the client.  It seems like it would be clearer and more resilient to use `convert_tags()` to explicitly pass it in the form the client expects.  Of course, much of this confusion could be addressed by only ever passing tag data in a single form (probably list since that's how we normalize it when loading the resources).

I still think it improves code readability to use `convert_tags()` where possible.

Reply:
Ah so I think that at one time API Gateway didn't have an augment and was returning tags in Dict form and that's why that condition was added in https://github.com/cloud-custodian/cloud-custodian/pull/5607. In #5681 augments were added though so not sure if it would still be a problem anymore.